### ISSUE TITLE
Fail to access watch later list on Bilibili

### ISSUE THREAD

**Issue by ischt** [yt-dlp/yt-dlp#8499]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.10.13** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

China

### Provide a description that is worded well enough to be understood

The current version can not access watch later list on Bilibili.
The solution should be simple, for example, just to replace the original URL [https://www.bilibili.com/list/watchlater?bvid=REALBVID] to [https://www.bilibili.com/video/REALBVID]

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.bilibili.com/list/watchlater?bvid=BV1kw411R7Jo']
[debug] User config "C:\Users\Tian\AppData\Roaming\yt-dlp\config": ['-o', 'C:\\Users\\Tian\\Downloads\\Streaming\\%(fulltitle)s.%(playlist_title)s.%(channel)s.%(display_id)s.%(uploader)s.%(uploader_id)s.%(upload_date)s.%(ext)s', '--cookies', 'C:\\Users\\Tian\\AppData\\Roaming\\yt-dlp\\bilibili.com_cookies.file']
[debug] Encodings: locale cp936, fs utf-8, pref cp936, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.10.13 [b634ba742] (pip)
[debug] Python 3.11.6 (CPython AMD64 64bit) - Windows-10-10.0.22621-SP0 (OpenSSL 3.0.11 19 Sep 2023)
[debug] exe versions: ffmpeg 5.0.1-full_build-www.gyan.dev (setts), ffprobe 5.0.1-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, certifi-2022.12.07, mutagen-1.47.0, sqlite3-3.42.0, websockets-12.0
[debug] Proxy map: {'http': 'http://127.0.0.1:7890', 'https': 'http://127.0.0.1:7890', 'ftp': 'http://127.0.0.1:7890'}
[debug] Loaded 1890 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
ERROR: Unable to obtain version info ([SSL: UNEXPECTED_EOF_WHILE_READING] EOF occurred in violation of protocol (_ssl.c:1006)); Please try again later or visit  https://github.com/yt-dlp/yt-dlp/releases/latest
[BilibiliPlaylist] Extracting URL: https://www.bilibili.com/list/watchlater?bvid=BV1kw411R7Jo
[BilibiliPlaylist] watchlater: Downloading webpage
ERROR: [BilibiliPlaylist] watchlater: Could not access playlist: -101 播单数 据请求失败; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "C:\Users\Tian\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\yt_dlp\extractor\common.py", line 715, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Tian\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\yt_dlp\extractor\bilibili.py", line 949, in _real_extract
    raise ExtractorError(f'Could not access playlist: {error_code} {error_message}')
```

**Comment by c-basalt** [comment]
This looks weird. Btw, are you able to use `yt-dlp --cookies cookies.file https://www.bilibili.com/watchlater` instead?

**Comment by grqz** [comment]
Closed by e439693f729daf6fb15457baea1bca10ef5da34d
